pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -146,22 +195,53 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 	found := &kubevirtv1.VirtualMachine{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: vmSpec.Name, Namespace: vmSpec.Namespace}, found)
 	if err != nil && errors.IsNotFound(err) {
+
+		// Create kubevirt VM from oVirt VM:
 		reqLogger.Info("Creating a new VM", "VM.Namespace", vmSpec.Namespace, "VM.Name", vmSpec.Name)
+		err = r.client.Create(context.TODO(), vmSpec)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
 
-		// Create kubevirt datavolume from oVirt VM disks:
+		// Secret with username/password for the image import:
+		if err = ensureDVSecretExists(r.client, instance, vmSpec.Namespace, ovirtSecretDataMap); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// CM containing CA for the image import:
+		if err = ensureDVConfigmapExists(r.client, instance, vmSpec.Namespace, ovirtSecretDataMap); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Import disks:
+		dvs := createDVmap(vm, ovirtSecretDataMap, instance.Namespace)
```

Comment:
It would be good to `SetControllerReference` on each `DataVolume` which would help us to get cancellation logic for free if VM import object was removed. We could unset it when the import is done.

Reply:
we'd like also to add a label to all of the resources created by the process (vm/dvs) :
`vmimport.v2v.kubevirt.io/tracker: ...` - but that can be added later on, based on the existence of the label.
The idea is that if that label was set for the vmimport cr, we should propagate it to the resources that were created by as a result (dvs secret/configmap seem less important in that context)